### PR TITLE
fix: add feature to block parallel ingestion from the same layer

### PR DESCRIPTION
**Pull request by razbroc**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: src/layers/models/layersManager.ts [hunk 1]
```diff
@@ -336,15 +336,20 @@ export class LayersManager {
...
       productType,
       isCleaned: false,
       shouldReturnTasks: false,
     };
+    const ingestionJobTypes = this.config.get<string[]>('forbiddenTypesForParallelIngesion');
     const jobs = await this.db.getJobs<Record<string, unknown>, ITaskParameters | IMergeTaskParams>(findJobParameters);
     jobs.forEach((job) => {
-      if (job.status == OperationStatus.IN_PROGRESS || job.status == OperationStatus.PENDING) {
+      if (
+        job.status == OperationStatus.IN_PROGRESS ||
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts]: This is a bug you need to check 
(job.status == OperationStatus.IN_PROGRESS || job.status == OperationStatus.PENDING) && ingestionJobTypes.indexOf(job.type) == notFoundIndex

File: src/layers/models/layersManager.ts [hunk 2]
```diff
@@ -336,15 +336,20 @@ export class LayersManager {
...
 
   private async validateJobNotRunning(productId: string, productType: ProductType): Promise<void> {
+    const notFoundIndex = -1;
     const findJobParameters: IFindJobsRequest = {
       resourceId: productId,
       productType,
       isCleaned: false,
       shouldReturnTasks: false,
     };
+    const ingestionJobTypes = this.config.get<string[]>('forbiddenTypesForParallelIngesion');
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:345]: check if this is being overriden from helm values

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good Job, Please mind my remarks 👍

File: src/layers/models/layersManager.ts (hunk 2 above)
Comment (razbroc) [reply src/layers/models/layersManager.ts:345]: values from helm are transferred fine :)

File: src/layers/models/layersManager.ts (hunk 1 above)
Comment (razbroc) [reply src/layers/models/layersManager.ts]: if the job for this layer is ingestion, and it's pending or in progress throw ConflictError.
can you explain further what you meant?

File: src/layers/models/layersManager.ts (hunk 2 above)
Comment (razbroc) [reply src/layers/models/layersManager.ts:345]: image

File: helm/values.yaml [hunk 3]
```diff
@@ -55,6 +55,9 @@ image:
 
 enableWMS: true
 env:
+  forbiddenTypesForParallelIngesion:
```
Comment (asafMasa) [inline helm/values.yaml:58]: add values to raster helm

File: src/layers/models/layersManager.ts (hunk 2 above)
Comment (razbroc) [reply src/layers/models/layersManager.ts:345]: checked it is good

File: helm/values.yaml (hunk 3 above)
Comment (razbroc) [reply helm/values.yaml:58]: https://github.com/MapColonies/raster-helm/pull/416

File: src/layers/models/layersManager.ts [hunk 4]
```diff
@@ -342,9 +342,10 @@ export class LayersManager {
...
     const jobs = await this.db.getJobs<Record<string, unknown>, ITaskParameters | IMergeTaskParams>(findJobParameters);
     jobs.forEach((job) => {
-      if (job.status == OperationStatus.IN_PROGRESS || job.status == OperationStatus.PENDING) {
+      if ((job.status == OperationStatus.IN_PROGRESS || job.status == OperationStatus.PENDING) && ingestionJobTypes.includes(job.type)) {
         const message = `Layer id: ${productId} product type: ${productType}, job is already running`;
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts]: `Layer id: ${productId} product type: ${productType}, conflicting job ${ingestionJobTypes[0]} is already running`

File: tests/integration/layers/layers.spec.ts [hunk 5]
```diff
@@ -687,7 +687,10 @@ describe('layers', function () {
   describe('Sad Path', function () {
     // All requests with status code 4XX-5XX
     it('should return 409 if rested layer is already being generated', async function () {
-      const jobs = [{ status: OperationStatus.FAILED }, { status: OperationStatus.IN_PROGRESS }];
+      const jobs = [
```
Comment (asafMasa) [inline tests/integration/layers/layers.spec.ts:722]: add a test for export jobType - that it's valid

File: tests/unit/layers/models/layersManager.spec.ts [hunk 6]
```diff
@@ -348,7 +348,7 @@ describe('LayersManager', () => {
       fileValidatorValidateExistsMock.mockResolvedValue(true);
       validateSourceDirectoryMock.mockResolvedValue(true);
       validateNotWatchDirMock.mockResolvedValue(true);
-      getJobsMock.mockResolvedValue([{ status: OperationStatus.PENDING }]);
+      getJobsMock.mockResolvedValue([{ status: OperationStatus.PENDING, type: 'Ingestion_New' }]);
```
Comment (asafMasa) [inline tests/unit/layers/models/layersManager.spec.ts:371]: add a test for export jobType - that it's valid

File: tests/integration/layers/layers.spec.ts [hunk 7]
```diff
@@ -687,7 +687,10 @@ describe('layers', function () {
...
+        { status: OperationStatus.FAILED, type: 'Ingestion_New' },
+        { status: OperationStatus.IN_PROGRESS, type: 'Ingestion_New' },
```
Comment (asafMasa) [inline tests/integration/layers/layers.spec.ts:724]: Check that we have a test for Aborted and Expired (ingestion) jobs

**Review by asafMasa — CHANGES_REQUESTED** [review]
Good Job - looking good 😎 
Please mind my remarks

File: src/layers/models/layersManager.ts (hunk 4 above)
Comment (razbroc) [reply src/layers/models/layersManager.ts]: changed

File: tests/integration/layers/layers.spec.ts (hunk 5 above)
Comment (razbroc) [reply tests/integration/layers/layers.spec.ts:722]: added

File: tests/integration/layers/layers.spec.ts (hunk 7 above)
Comment (razbroc) [reply tests/integration/layers/layers.spec.ts:724]: added

File: tests/unit/layers/models/layersManager.spec.ts (hunk 6 above)
Comment (razbroc) [reply tests/unit/layers/models/layersManager.spec.ts:371]: added

File: tests/integration/layers/layers.spec.ts [hunk 8]
```diff
@@ -705,6 +737,44 @@ describe('layers', function () {
       expect(createTasksMock).toHaveBeenCalledTimes(0);
     });
 
+    it('should return 409 if rested layer is already being generated while also having aborted job', async function () {
```
Comment (asafMasa) [inline tests/integration/layers/layers.spec.ts]: rested -> rename to tested

Comment (asafMasa) [reply tests/integration/layers/layers.spec.ts]: "generated" change to "In-Progress"

File: tests/integration/layers/layers.spec.ts [hunk 9]
```diff
@@ -705,6 +737,44 @@ describe('layers', function () {
...
+      expect(response.status).toBe(httpStatusCodes.CONFLICT);
+      expect(getJobsMock).toHaveBeenCalledTimes(1);
+      expect(getHighestLayerVersionMock).toHaveBeenCalledTimes(0);
+      expect(mapExistsMock).toHaveBeenCalledTimes(0);
+      expect(catalogExistsMock).toHaveBeenCalledTimes(0);
+      expect(createLayerJobMock).toHaveBeenCalledTimes(0);
+      expect(createTasksMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('should return 409 if rested layer is already being generated while also having expired job', async function () {
```
Comment (asafMasa) [inline tests/integration/layers/layers.spec.ts]: rested -> "tested"

**Review by asafMasa — CHANGES_REQUESTED** [review]

**Review by asafMasa — APPROVED** [review]

**Pull request merged**